src/HDWallet.cpp

```diff
@@ -32,28 +34,62 @@ HDNode getMasterNode(const HDWallet& wallet, TWCurve curve);
 const char* curveName(TWCurve curve);
 } // namespace
 
+std::mutex mnemonicGenerateGuard;
+std::mutex mnemonicFromDataGuard;
```

Comment:
do we really need this? just mention this method is not thread safe?

Reply:
There are multi-threaded HDWallet tests, which failed (they produced incorrect result but there was no explicit error check before).  We have these tests for a reason.  If the android app side is ok with non-thread-safe, I remove.

Reply:
I made changes in trezor-crypto, to make methods thread safe.

Reply:
Good